pkgs/development/libraries/science/math/ecos/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "ecos-${version}";
+  version = "2.0.6";
+
+  src = fetchFromGitHub {
+    owner = "embotech";
+    repo = "ecos";
+    rev = "v${version}";
+    sha256 = "11v958j66wq30gxpjpkgl7n3rvla845lygz8fl39pgf1vk9sdyc7";
+  };
+
+  doCheck = true;
+
+  buildPhase = ''
+    make all shared
+  '';
+
+  checkPhase = ''
+    make test
+    ./runecos
+  '';
+
```

Comment:
The upstream makefile doesn't ship a proper `make check` or `make install`, but these get the job done.